### PR TITLE
Simmy API review Part 2 - Prefer Chaos over Monkey

### PR DESCRIPTION
**Pull request by peter-csala**
# Pull Request

## The issue or feature being addressed

- #1901 
- #1909 

## Details on the issue fix or feature implementation

- Renamed the base classes
  - `MonkeyStrategy` >> `ChaosStrategy`
  - `MonkeyStrategyOptions` >> `ChaosStrategyOptions`
- Renamed Behavior classes
  - `BehaviorChaosStrategy` to `ChaosBehaviorStrategy`
  - `BehaviorStrategyOptions` >> `ChaosBehaviorStrategyOptions`
  - `BehaviorConstants` >> `ChaosBehaviorConstants`  
  - `BehaviorPipelineBuilderExtensions` >> `ChaosBehaviorPipelineBuilderExtensions`
- Renamed Fault classes
  - `FaultChaosStrategy` to `ChaosFaultStrategy`
  - `FaultStrategyOptions` >> `ChaosFaultStrategyOptions`
  - `FaultConstants` >> `ChaosFaultConstants`  
  - `FaultPipelineBuilderExtensions` >> `ChaosFaultPipelineBuilderExtensions`  
- Renamed Latency classes
  - `LatencyChaosStrategy` to `ChaosLatencyStrategy`
  - `LatencyStrategyOptions` >> `ChaosLatencyStrategyOptions`
  - `LatencyConstants` >> `ChaosLatencyConstants`  
  - `LatencyPipelineBuilderExtensions` >> `ChaosLatencyPipelineBuilderExtensions`   
  - Renamed Outcome classes
  - `OutcomeChaosStrategy` to `ChaosOutcomeStrategy`
  - `OutcomeStrategyOptions` >> `ChaosOutcomeStrategyOptions`
  - `OutcomeConstants` >> `ChaosOutcomeConstants`  
  - `OutcomePipelineBuilderExtensions` >> `ChaosOutcomePipelineBuilderExtensions`   
- Renamed `AddChaosResult` to `AddChaosOutcome`
- Updated related docs as well

## Extra
- Replaced `behaviour` spelling to `behavior` in `ChaosBehaviorStrategyTests`
- Moved `userDelegateExecuted` and `onXYZInjectedExecuted` method variables to class level in several test classes

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/chaos/index.md [hunk 1]
```diff
@@ -25,9 +25,9 @@ Using Polly helps introduce resilience to a project, but we don't want to have t
 * A way to revert easily, to control the blast radius.
 * To be production grade, to run this in a production or near-production system with automation.
 
-## Chaos strategies (a.k.a Monkey strategies)
+## Chaos strategies
 
-Chaos strategies (or Monkey strategies as we call them) are in essence a [Resilience strategy](../strategies/index.md#built-in-strategies), which means, as a *Resilience Strategy* is the minimum unit of resilience for Polly, a *Chaos Strategy* is the minimum unit of chaos for Simmy.
```
Comment (martincostello) [inline docs/chaos/index.md:30]: Maybe leave a note that that's what they were referred to previously?

**Comment by martincostello** [comment]
I also prefer option A.

File: docs/chaos/index.md (hunk 1 above)
Comment (peter-csala) [reply docs/chaos/index.md:30]: Done

**Review by martintmk — APPROVED** [review]
Looks fine, great job!

**Comment by martintmk** [comment]
@martincostello Is this ok to merge?

**Review by martincostello — APPROVED** [review]

**Pull request merged**